packages/components/src/focal-point-picker/index.native.js

```diff
@@ -0,0 +1,178 @@
+import { useRef, useState, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import FocalPoint from './focal-point';
+import Tooltip from './tooltip';
+import styles from './style.scss';
+
+const MIN_POSITION_VALUE = 0;
+const MAX_POSITION_VALUE = 100;
+
+export default function FocalPointPicker( props ) {
+	const { focalPoint, onChange, shouldEnableBottomSheetScroll, url } = props;
+
+	const [ containerSize, setContainerSize ] = useState( null );
+	const [ sliderKey, setSliderKey ] = useState( 0 );
+	const [ tooltipVisible, setTooltipVisible ] = useState( true );
+	const [ displayPlaceholder, setDisplayPlaceholder ] = useState( true );
+
+	const pan = useRef( new Animated.ValueXY() ).current;
+
+	// Set initial cursor poition
+	if ( containerSize ) {
+		pan.setValue( {
+			x: focalPoint.x * containerSize.width,
+			y: focalPoint.y * containerSize.height,
+		} );
+	}
+
+	const panResponder = useMemo(
+		() =>
+			PanResponder.create( {
+				onStartShouldSetPanResponder: () => true,
+				onStartShouldSetPanResponderCapture: () => true,
+				onMoveShouldSetPanResponder: () => true,
+				onMoveShouldSetPanResponderCapture: () => true,
+
+				onPanResponderGrant: ( event ) => {
+					shouldEnableBottomSheetScroll( false );
+					const { locationX: x, locationY: y } = event.nativeEvent;
+					pan.setValue( { x, y } ); // Set cursor to tap origin
+					pan.setOffset( { x: pan.x._value, y: pan.y._value } );
+				},
+				// Move cursor to match delta drag
+				onPanResponderMove: Animated.event( [
+					null,
+					{ dx: pan.x, dy: pan.y },
+				] ),
+				onPanResponderRelease: ( event ) => {
+					shouldEnableBottomSheetScroll( true );
+					setTooltipVisible( false );
+					pan.flattenOffset();
+					const { locationX: x, locationY: y } = event.nativeEvent;
+					setPosition( {
+						x: x / containerSize?.width,
+						y: y / containerSize?.height,
+					} );
+					// Slider (child of RangeCell) is uncontrolled, so we must increment a
+					// key to re-mount and sync the pan gesture values to the sliders
+					// https://git.io/JTe4A
+					setSliderKey( ( prevState ) => prevState + 1 );
+				},
+			} ),
+		[ containerSize ]
+	);
+
+	function setPosition( { x, y } ) {
+		onChange( ( prevState ) => ( {
+			...prevState,
+			...( x ? { x } : {} ),
+			...( y ? { y } : {} ),
+		} ) );
+	}
+
+	return (
+		<View style={ styles.container }>
+			<View style={ [ styles.media ] }>
+				<View
+					{ ...panResponder.panHandlers }
+					onLayout={ ( event ) => {
+						const { height, width } = event.nativeEvent.layout;
+
+						if (
+							width !== 0 &&
+							height !== 0 &&
+							( containerSize?.width !== width ||
+								containerSize?.height !== height )
+						) {
+							setContainerSize( { width, height } );
+						}
+					} }
+					style={ styles.imageContainer }
+				>
+					<Image
+						height="100%"
+						url={ url }
+						style={ [
+							displayPlaceholder && styles.imagePlaceholder,
```

Comment:
What do you think if we put this in a const? Like `imagePlaceholderStyles`.

Reply:
@geriux yeah, that makes sense to me.